File: snippets/fsharp/VS_Snippets_CLR_System/system.Func~1/fs/Delegate.fs
module Delegate

// <Snippet1>
open System.IO

type WriteMethod = delegate of unit -> bool

type OutputTarget() =
    member _.SendToFile() =
        try
            let fn = Path.GetTempFileName()
            use sw = new StreamWriter(fn)
            sw.WriteLine "Hello, World!"
            true
        with _ ->
            false

let output = new OutputTarget()
let methodCall = WriteMethod output.SendToFile
if methodCall.Invoke() then
    printfn "Success!"
else
    printfn "File write operation failed."

// </Snippet1>

File: snippets/csharp/System/FuncTResult/Overview/Delegate.cs
// <Snippet1>
using System;
using System.IO;

delegate bool WriteMethod();

public class CustomDelegateExample
{
    public static void Run()
    {
        CustomDelegateOutputTarget output = new();
        WriteMethod methodCall = output.SendToFile;
        if (methodCall())
            Console.WriteLine("Success!");
        else
            Console.WriteLine("File write operation failed.");
    }
}

public class CustomDelegateOutputTarget
{
    public bool SendToFile()
    {
        try
        {
            string fn = Path.GetTempFileName();
            using StreamWriter sw = new(fn);
            sw.WriteLine("Hello, World!");
            return true;
        }
        catch
        {
            return false;
        }
    }
}
// </Snippet1>
